tt_metal/jit_build/build.cpp

```diff
@@ -582,6 +582,15 @@ void JitBuildState::compile_one(
         settings->process_compile_time_args([&defines](int i, uint32_t value) {
             defines += "-DKERNEL_COMPILE_TIME_ARG_" + to_string(i) + "=" + to_string(value) + " ";
         });
+
+        std::stringstream args;
+        for (size_t i = 0; i < static_cast<const Kernel*>(settings)->compile_time_args().size(); i++) {
+            if (i != 0) {
+                args << ",";
+            }
+            args << static_cast<const Kernel*>(settings)->compile_time_arg(i);
+        }
+        defines += "-DKERNEL_COMPILE_TIME_ARGS=\\(int[]\\){" + args.str() + "} ";
```

Comment:
Was there a reason for changing this to c-style array being directly defined here? Was there a problem with the other approach in the originally linked godbolt?

i.e.
```
#include <array>
#include <cstdint>

template <class T, class... Ts>
constexpr auto make_array(Ts... values) -> std::array<T, sizeof...(Ts)> {
    return {T(values)...};
}

inline constexpr auto kernel_compile_time_args{
    make_array<std::uint32_t>(KERNEL_COMPILE_TIME_ARGS),
};
```

This can be defined in `tt_metal/hw/inc/compile_time_args.h` and included by `dataflow_api.h`?